src/net_processing.cpp

```diff
@@ -3723,32 +3719,32 @@ void ProcessMessage(
  */
 bool PeerLogicValidation::MaybeDiscourageAndDisconnect(CNode& pnode)
 {
-    NodeId peer_id{pnode.GetId()};
+    const NodeId peer_id{pnode.GetId()};
     {
         LOCK(cs_main);
-        CNodeState &state = *State(peer_id);
+        CNodeState& state = *State(peer_id);
 
         // There's nothing to do if the m_should_discourage flag isn't set
         if (!state.m_should_discourage) return false;
 
-        // Reset m_should_discourage
         state.m_should_discourage = false;
     } // cs_main
 
     if (pnode.HasPermission(PF_NOBAN)) {
-        // Peer has the NOBAN permission flag - log but don't disconnect
+        // We never disconnect or discourage peers for bad behavior if they have the NOBAN permission flag 
```

Comment:
dd9b305 whitespace linter needs appeasing (rm space at EOL)
```suggestion
        // We never disconnect or discourage peers for bad behavior if they have the NOBAN permission flag
```

Reply:
am reviewing anyway, so can wait on this until the review, if you like

Reply:
Fixed